src/controllers/engine/controllerengine.cpp

```diff
@@ -136,6 +143,9 @@ QJSValue ControllerEngine::wrapFunctionCode(const QString& codeSnippet,
                                 codeSnippet + ")(" + wrapperArgs + "); })";
 
         wrappedFunction = evaluateCodeString(wrappedCode);
+        if (wrappedFunction.isError()) {
+            showScriptExceptionDialog(wrappedFunction);
```

Comment:
Maybe we should make `evaluateCodeString` do this check instead?

Reply:
Oh, that's how it was... why did you remove that?

Reply:
I moved `showScriptExceptionDialog(...)` outside because now we don't always call it when we evaluate a code string. In `beginTimer(...)` and `connectControl` we want to throw a custom error. If we don't do so, the line number is incorrect.

Note that since `makeConnection(...)` does not accept strings, any errors in the callback are caught earlier when parsing the code. So if we remove the possibility to pass strings to `beginTimer(...)` and `connectControl` we don't need this change. But we brake compatibility with old scripts.

Reply:
Hmm, that makes sense, but I think I'd rather have prettier C++ code with less chance for introducing inconsistent error handling accidentally later than better error handling for deprecated functionality.

Reply:
About `beginTimer`, all the script we ship use it with a function, so we can make it stop supporting strings. But `connectControl` is used a lot in our scripts with strings. I think it's ok if we leave it like this.